sale_coupon_domain_product_discount/models/coupon_reward.py

```diff
@@ -35,7 +35,7 @@ def _compute_discount_specific_product_ids(self):
         Otherwise it just masks the stored field"""
         domain_product = self.env.context.get("promo_domain_product")
         if domain_product:
-            self.discount_specific_product_ids = self.env["product.product"]._search(
+            self.discount_specific_product_ids = self.env["product.product"].search(
```

Comment:
Why?

Reply:
Using "_search" method directly was giving me an error about wrong value of a field.  I tested in this way and worked, but I didn't investigate more the error.

Reply:
Well, performance is crucial here so you should find out what that error might be due to.